### DISCUSSION TITLE
Get Document Auto-ID for Firestore

### DISCUSSION THREAD

**Discussion by JediThumper** [SchneiderInfosystems/FB4D#165] [Q&A]
Hi,

I'm sure, I've done this before, but I can't remember.

How do I get an auto-id for a new document?

Thanks

Alan

PS

Compiled with no issues on D12

**Comment by JediThumper** [comment]
Found it :)
Use Database.CreateDocument(Synchronous) 
then retrieve the DocumentName.

Alan

**Reply by SchneiderInfosystems** [reply]
In addition to the mentioned server side approach the unit FB4D.Helpers offers a further client side approach by using `TFirebaseHelpers.CreateAutoID(IDKind: TIDKind): string`
For the Firestore Database there are two ID kinds for different purposes:
* FSID: Generates a random ID
* PushID: Generates a random ID beginning with the current timestamp allows fast sorting from oldest the newest document by the ID

**Comment by JediThumper** [comment]
Thanks for the additional information Christoph.
So many little gems I'm still learning.
